File: app/src/main/java/club/anlan/lanlife/lanapp/receiver/LocationAlarmReceiver.java
package club.anlan.lanlife.lanapp.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import club.anlan.lanlife.lanapp.service.LocationAlarmService;

/**
 * 定位 广播接收器
 *
 * @author lan
 * @version 1.0
 * @date 2021/9/12 16:44
 */
public class LocationAlarmReceiver extends BroadcastReceiver {

    private static final String TAG = "LocationAlarmReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        //循环启动 Service
        Log.e(TAG, "收到");
        Intent i = new Intent(context, LocationAlarmService.class);
        context.startService(i);
    }
}

File: app/src/main/java/club/anlan/lanlife/app/receiver/LocationAlarmReceiver.kt
package club.anlan.lanlife.app.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import club.anlan.lanlife.app.service.LocationService

/**
 *
 * 定位自启动
 *
 * @author lan
 * @date 2022/11/9 0:24
 * @version 1.0
 */
class LocationAlarmReceiver : BroadcastReceiver() {

    private val TAG = "LocationAlarmReceiver"

    override fun onReceive(context: Context?, intent: Intent?) {
        // 循环启动 Service
        val i = Intent(context, LocationService::class.java)
        context?.startService(i)
    }
}